### PR TITLE
SUS-837: Remove Phalanx fallback

### PR DESCRIPTION
**Pull request by themech**

### CODE REVIEW

File: extensions/wikia/PhalanxII/models/PhalanxUserModel.php [hunk 1]
```diff
@@ -42,12 +42,44 @@ public function getText() {
...
+                $result = wfMsg( 'phalanx-user-block-withreason-exact', $reason );
+            } elseif ( $isBlockIP ) {
+                $result = wfMsg( 'phalanx-user-block-withreason-ip', $reason );
+            } else {
+                $result = wfMsg( 'phalanx-user-block-withreason-similar', $reason );
+            }
+        } else {
+            // no reason in block data, so use preexisting no-param worded versions
+            if ( $isExact ) {
+                $result = wfMsg( 'phalanx-user-block-reason-exact' );
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/models/PhalanxUserModel.php:67]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: extensions/wikia/PhalanxII/models/PhalanxUserModel.php [hunk 2]
```diff
@@ -42,12 +42,44 @@ public function getText() {
...
+            } elseif ( $isBlockIP ) {
+                $result = wfMsg( 'phalanx-user-block-reason-ip' );
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/models/PhalanxUserModel.php:69]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: extensions/wikia/PhalanxII/models/PhalanxUserModel.php [hunk 3]
```diff
@@ -42,12 +42,44 @@ public function getText() {
...
+            } else {
+                $result = wfMsg( 'phalanx-user-block-reason-similar' );
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/models/PhalanxUserModel.php:71]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: extensions/wikia/PhalanxII/models/PhalanxUserModel.php [hunk 4]
```diff
@@ -42,12 +42,44 @@ public function getText() {
...
+     * @param String $reason Optional reason that came with the block
+     * @param bool $isExact Set to true for exact block
+     * @param bool $isBlockIP Set to true for IP blocks
+     * @return String Translated message with optional reason details
+     */
+    protected function getBlockReasonMessage( $reason, $isExact, $isBlockIP ) {
+        if ( $reason ) {
+            // a reason was given
+            if ( $isExact ) {
+                $result = wfMsg( 'phalanx-user-block-withreason-exact', $reason );
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/models/PhalanxUserModel.php:58]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: extensions/wikia/PhalanxII/models/PhalanxUserModel.php [hunk 5]
```diff
@@ -42,12 +42,44 @@ public function getText() {
...
+            } elseif ( $isBlockIP ) {
+                $result = wfMsg( 'phalanx-user-block-withreason-ip', $reason );
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/models/PhalanxUserModel.php:60]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

File: extensions/wikia/PhalanxII/models/PhalanxUserModel.php [hunk 6]
```diff
@@ -42,12 +42,44 @@ public function getText() {
...
+            } else {
+                $result = wfMsg( 'phalanx-user-block-withreason-similar', $reason );
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/models/PhalanxUserModel.php:62]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 7 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PhalanxModel.php#L152](https://github.com/Wikia/app/blob/dc6db7635ad4220ef4d8b2d06d8dc46f919b2df1/extensions/wikia/PhalanxII/models/PhalanxModel.php#L152): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by macbre — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**